coin/coin.go

```diff
@@ -260,22 +260,23 @@ func (c *Coin) Clone() *Coin {
 // so you may want to make other checks in your business
 // logic
 func (c Coin) Validate() error {
+	var err error
 	if !IsCC(c.Ticker) {
-		return errors.Wrapf(errors.ErrCurrency, "invalid currency: %s", c.Ticker)
+		err = errors.Wrapf(errors.ErrCurrency, "invalid currency: %s", c.Ticker)
```

Comment:
I think it's better to use `errors.Append` here as well for consistency. If someone will add an error before this check, it can cause a bug.

Reply:
yeah, I was thinking about this, but at the same time some tests would fail because of that too! 
I'm still waiting to see if this even makes sense to merge tbh, as I'm not happy with the fact that we can't tell which field is empty in a test when we receive `ErrIsEmpty`, for example.

Reply:
> I'm still waiting to see if this even makes sense to merge tbh, as I'm not happy with the fact that we can't tell which field is empty in a test when we receive ErrIsEmpty, for example.

We couldn't see that before either.
Now, at least we see two fields are empty and a coin is mis-encoded.
And ideally if we read the logs, they tell us (a developer) which field is messed up. Sure, we cannot *programatically* determine fields, but since we usually Wrap with the field name in validation, this should be good enough for us to fix our code faster.